src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,154 @@
+        assertThrows(RegistrationException.class, () -> registrationService.register(
+                new User(LOGIN_YURI, null, MINIMUM_AGE)));
+    }
+
+    @Test
+    void register_NullAgeValue_NotOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(
+                new User(LOGIN_YURI, CORRECT_PASSWORD, null)));
+    }
+
+    @Test
+    void register_correctEge_Ok() {
+        registrationService.register(new User(LOGIN_YURI, CORRECT_PASSWORD, MINIMUM_AGE));
+        registrationService.register(new User(LOGIN_DASHA, CORRECT_PASSWORD, OVER_MINIMUM_AGE));
+        assertTrue(people.contains(new User(LOGIN_YURI, CORRECT_PASSWORD, MINIMUM_AGE)));
+        assertTrue(people.contains(new User(LOGIN_DASHA, CORRECT_PASSWORD, OVER_MINIMUM_AGE)));
+        assertEquals(EXPECTED_TWO, people.size());
+    }
+
+    @Test
+    void register_negativeAge_NotOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(
+                new User(CORRECT_LOGIN, CORRECT_PASSWORD, NEGATIVE_AGE)));
+    }
+
+    @Test
+    void register_zeroAge_NotOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(
+                new User(CORRECT_LOGIN, CORRECT_PASSWORD, ZERO_AGE)));
+    }
+
+    @Test
+    void register_belowMinimumAge_NotOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(
+                new User(CORRECT_LOGIN, CORRECT_PASSWORD, BELOW_MINIMUM_AGE)));
+    }
+
+    @Test
+    void register_correctLogin_Ok() {
+        registrationService.register(new User(LOGIN_YURI, CORRECT_PASSWORD, MINIMUM_AGE));
+        registrationService.register(new User(LOGIN_DASHA, CORRECT_PASSWORD, MINIMUM_AGE));
+        registrationService.register(new User(LOGIN_NAZAR, CORRECT_PASSWORD, MINIMUM_AGE));
+        registrationService.register(new User(
+                CORRECT_LOGIN_SIX_LETTERS, CORRECT_PASSWORD, MINIMUM_AGE));
+        assertTrue(people.contains(new User(LOGIN_YURI, CORRECT_PASSWORD, MINIMUM_AGE)));
+        assertTrue(people.contains(new User(LOGIN_DASHA, CORRECT_PASSWORD, MINIMUM_AGE)));
+        assertTrue(people.contains(new User(LOGIN_NAZAR, CORRECT_PASSWORD, MINIMUM_AGE)));
+        assertTrue(people.contains(new User(
+                CORRECT_LOGIN_SIX_LETTERS, CORRECT_PASSWORD, MINIMUM_AGE)));
+        assertEquals(EXPECTED_FOUR, people.size());
+    }
+
+    @Test
+    void register_incorrectLogin_NotOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(
+                new User(INCORRECT_LOGIN, CORRECT_PASSWORD, MINIMUM_AGE)));
+    }
+
+    @Test
+    void register_incorrectLoginEmptyLine_NotOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(
+                new User(EMPTY_LINE, CORRECT_PASSWORD, MINIMUM_AGE)));
+    }
+
+    @Test
+    void register_correctPassword_Ok() {
```

Comment:
This test should be split into separate tests for different correct password scenarios to maintain test independence. Each test should have a clear name indicating the scenario being tested.